### PR TITLE
Cleanup test

### PR DESCRIPTION
**Pull request by jaemolihm**
Cleanup files in `tests/` to make them more organized and make flake8 work.

* Make constant fixtures constant variables
* Reorganize fixtures to `common.py`, `common_systems.py` and `common_comparers.py`
* Import all common fixtures in conftest.py so that they do not need to be imported in individual tests
* Changed test reference file `tests/reference/path_tab.pickle` : the key for energy data is changed from `('E', None)` to `('Energy', '')` to be consistent with `__tabulate.py`. Data values are not changed.

Assumes #138 is merged.

### TODO
* [x] Move `README_TESTS.rst` to `tests/` (?)
* [x] Add explanation on how to add new things
  * Objects that can be shared by multiple tests should go to `common_*.py`.
  * Do not use fixtures for simple constants.
  * Use fixtures only if 1) it takes a long time or should not be run multiple times, or 2) test error (assert) can be raised during the execution of that part.
* [x] Setup flake8 for tests

### CODE REVIEW

**Comment by jaemolihm** [comment]
I added a section "Recommended styles" in `tests/README.rst` that explains the style change.

For example, in #141, I suggest the following changes:
* Move new models and new systems to `common_systems.py`
* Rename `ChiralModelLeft` to `model_chiral_left` (or `_left_handed`) and make them simple constants: see https://github.com/wannier-berri/wannier-berri/pull/140/files#diff-1094acc3896f0795bdef72c1398a80e753f7ef7b937d170c77ffa1d147bc4233R219
* Import `system_Chiral_*` in conftest.py: https://github.com/wannier-berri/wannier-berri/pull/140/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R8-R14
* Make `calculators_*` constants not fixtures.

I suggest you merge #141 first, and then I will implement these changes and solve merge conflicts in this PR, and then merge this PR.

**Review by stepan-tsirkin — COMMENTED** [review]
Why was it necessary to update the `path_tab.pickle` file? Could you comment on it in the PR description.

**Comment by jaemolihm** [comment]
> Why was it necessary to update the `path_tab.pickle` file? Could you comment on it in the PR description.

I did so. It was to change the key for energy data from `('E', None)` to `('Energy', '')` so that it is consistent with `__tabulate.py`

File: tests/test_cluster.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import os
 import pytest
 
-from conftest import REF_DIR, OUTPUT_DIR
+from common import REF_DIR, OUTPUT_DIR
```
[the remark this answers is not in the document]
Comment (jaemolihm) [reply tests/test_cluster.py:4]: That holds only for fixtures, not simple variables or functions.

**Review by stepan-tsirkin — APPROVED** [review]
Now ok

**Pull request merged**